programs/svm-spoke/src/instructions/fill.rs

```diff
@@ -114,15 +118,23 @@ pub fn fill_relay<'info>(
         _ => FillType::FastFill,
     };
 
-    // Relayer must have delegated output_amount to the state PDA
+    // Verify delegate PDA
+    let seed_hash = derive_seed_hash(&(FillSeedData { relay_hash, repayment_chain_id, repayment_address }));
+    let (pda, bump) = Pubkey::find_program_address(&[b"delegate", &seed_hash], &ctx.program_id);
+    if ctx.accounts.delegate.key() != pda {
+        return err!(SvmError::InvalidDelegatePda);
+    }
```

Comment:
better move this logic inside `transfer_from` by passing `seed_hash` as we have the same logic for deposits